src/main.rs

```diff
@@ -70,36 +42,50 @@ fn main() {
 }
 
 fn run() -> Result<(), MainError> {
-    let args: Vec<String> = std::env::args().collect();
-    match &*args[1] {
-        "run" | "run-script" => {
-            let manifest = "package.json"
+    let cli = Cli::parse();
+    match cli.command {
+        cli::Command::Run(args) => {
+            let mut cwd = env::current_dir().expect("Couldn't find the current working directory");
+            if cli.workspace_root {
+                cwd = workspace::find_workspace_root(&cwd)?;
+            }
+            let manifest_path = cwd.join("package.json");
+            let manifest = manifest_path
                 .pipe(File::open)
                 .map_err(MainError::from_dyn)?
                 .pipe(serde_json::de::from_reader::<_, NodeManifest>)
                 .map_err(MainError::from_dyn)?;
-            let name = args[2].clone();
-            if let Some(command) = manifest.scripts.get(&name) {
+            if let Some(command) = manifest.scripts.get(&args.script) {
                 eprintln!("> {:?}", command);
-                run_script(name, command.clone())
+                run_script(args.script, command.clone(), &cwd)
             } else {
-                PnError::MissingScript { name }
+                PnError::MissingScript { name: args.script }
                     .pipe(MainError::Pn)
                     .pipe(Err)
             }
         }
-        "install" | "i" | "update" | "up" => pass_to_pnpm(&args[1..]),
-        _ => pass_to_sub(args[1..].join(" ")),
+        cli::Command::Install(passed_trough_args) => {
+            let mut args = passed_trough_args.args;
+            args.insert(0, "install".into());
+            pass_to_pnpm(&args)
+        }
+        cli::Command::Update(passed_trough_args) => {
+            let mut args = passed_trough_args.args;
+            args.insert(0, "update".into());
+            pass_to_pnpm(&args)
+        }
```

Comment:
I mean this, I don't want to duplicate this for all the pnpm commands. Is there a more compact way to fallback to pnpm commands?

Reply:
I will try to refactor this in my editor.

Reply:
> Is there a more compact way to fallback to pnpm commands?

I just noticed: Initially, I didn't define `Install` or `Update` because I thought everything else is just going to be passed to `pnpm`. But since you separate them from `Other`, the code is going to be a bit complicated. I have an idea about how to achieve it (just define 1 more struct and 1 more enum), but I'll wait until you resolve the current suggestions before I open my editor.

Reply:
I think I have changed everything that you suggested already.

Reply:
OK. I will open my editor now.